### PR TITLE
feat(MySql): Add MCP support for Aurora MySQL

### PR DESCRIPTION
**Pull request by kennthhz**
Fixes

## Summary

Add MCP support for Aurora MySQL

### Changes

Add MCP support for Aurora MySQL
Support read only transactions 
Reject risky queries

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x ] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [ x] I have performed a self-review of this change
* [ x] Changes have been tested
* [ x] Changes are documented

Is this a breaking change? (Y/N)

**RFC issue number**:

Checklist:

* [ x] Migration process documented
* [ x] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

### CODE REVIEW

**Review by awslabs-mcp — COMMENTED** [review]
## MySQL MCP Server PR Review

I've conducted a thorough review of this PR which adds Aurora MySQL MCP server support. Overall, this is a solid contribution that follows many of the established patterns for MCP servers. Here's my assessment:

### Strengths
- Comprehensive SQL injection protection with robust pattern detection
- Good implementation of readonly mode for safe database access
- Strong test coverage with tests for various SQL patterns and edge cases
- Clear documentation in README.md with installation and configuration instructions
- Proper error handling and logging
- Appropriate async/await patterns for database operations

### Workflow Run
- Run ID: 15176895586
- Most checks are passing, however there is a failing "fail-by-label" check. This appears to be related to labels on the PR.

### Recommendations for Improvement
1. Consider creating a dedicated `models.py` file with proper Pydantic models for requests/responses
2. Move global error message constants to a separate `consts.py` file
3. Add more detailed instructions in the MCP initialization (`instructions` parameter)
4. Add timeouts for long-running queries
5. Implement a maximum result set size limit
6. Replace the type ignore comment on line 190 with a proper Pydantic return type

These suggestions align with the MCP design guidelines and would improve code organization, safety, and documentation.

**Review by awslabs-mcp — APPROVED** [review]
## Final Review Decision: APPROVE

I'm approving this pull request to add MySQL MCP server support. The implementation is solid, well-tested, and follows AWS security best practices with its SQL injection protection and readonly mode configuration.

While I've recommended some improvements in my previous review comment regarding code organization and Pydantic models, these are enhancements rather than blockers. The current implementation:

1. Correctly implements the MCP server pattern
2. Provides excellent security features
3. Has comprehensive test coverage
4. Follows AWS authentication best practices
5. Has clear documentation

Great work on this contribution to the MCP ecosystem, adding support for Aurora MySQL\!

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

**Review by krokoko — APPROVED** [review]

File: src/mysql-mcp-server/tests/test_server.py [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+    monkeypatch.setattr('awslabs.mysql_mcp_server.server.mcp.run', lambda: None)
+
+    # This test of main() will succeed in parsing parameters and create connection object.
+    # However, since connection object is not boto3 client with real credential, the validate of connection will fail and cause system exit
+    with pytest.raises(SystemExit) as excinfo:
+        main()
+    assert excinfo.value.code == 1
+
+
+if __name__ == '__main__':
```
Comment (dotmanila) [inline src/mysql-mcp-server/tests/test_server.py:736]: Do you think this is better enforced by requiring read-only accounts? For instance, on these coverage we need to add `CALL` and `SELECT some_mutating_function_name()`. If we do the former, the MCP server will validate the privileges at start and if it is not read-only, abort.
